### PR TITLE
fix: added log for invalid returned data from functions

### LINKED ISSUE #270: Nothing fails/logs when a Glee Function returns an unknown structure
#### Describe the bug
Glee Functions should either return `undefined` or a well-known object containing either `send` or `reply` keys. If I return something different, say an Array —for instance—, Glee doesn't complain in any way and therefore it's hard to debug what's wrong in your code.

#### How to Reproduce
Go change the [returned value here](https://github.com/asyncapi/glee/blob/master/examples/dummy/functions/onUserSignedUp.ts#L4) to an array and see nothing happens. The code will not work but it also doesn't complain.

#### Expected behavior
It should either fail or log a warning telling us the returned structure is not expected and that we should return either `undefined` or the [well-known structure](https://github.com/asyncapi/glee/blob/master/docs/functions.md).

### PR DESCRIPTION
**Pull request by sudoshreyansh**
**Description**

- added check and log for invalid data returned from functions

**Related issue(s)**
Fixes #270

### CODE REVIEW

**Review by fmvilas — CHANGES_REQUESTED** [review]
I think we should do better here instead of just checking for the existence of `send` or `reply`. What if I return an object like this:

```js
return {
  send: 10,
}
```

It may be a good idea to use JSON Schema for this and we already have `ajv` installed as a dependency :)

**Comment by sudoshreyansh** [comment]
Ok I have made the changes.

@fmvilas do you want to log the errors too while validating the returned value or just the invalid message?

File: src/lib/functions.ts [hunk 1]
```diff
@@ -6,13 +6,46 @@ import { logWarningMessage } from './logger.js'
...
 import { pathToFileURL } from 'url'
 
 interface FunctionInfo {
   run: GleeFunction,
 }
 
+const OutboundMessageSchema = {
+  type: 'object',
+  properties: {
+    payload: { type: ['string', 'object'] },
```
Comment (fmvilas) [inline src/lib/functions.ts:19]: Payload can be anything so I'd just make it like this:

```suggestion
    payload: {},
```

File: src/lib/functions.ts [hunk 2]
```diff
@@ -6,13 +6,46 @@ import { logWarningMessage } from './logger.js'
...
+    headers: {
+      type: 'object',
+      propertyNames: { type: 'string' },
+      additionProperties: { type: 'string' }
```
Comment (fmvilas) [inline src/lib/functions.ts:23]:
```suggestion
      additionalProperties: { type: 'string' }
```

File: src/lib/functions.ts [hunk 3]
```diff
@@ -54,10 +87,17 @@ export async function trigger({
 }) {
   try {
     const res = await functions.get(operationId).run(gleeMessageToFunctionEvent(message, app))
-    let valid = res === undefined
+    const { isValid } = validateData(res, FunctionReturnSchema)
+
+    if ( !isValid ) {
+      logWarningMessage(`Function ${operationId} returned invalid data`, {
+        highlightedWords: [operationId]
+      })
```
Comment (fmvilas) [inline src/lib/functions.ts:95]:
```suggestion
      const err = new GleeError({
        humanReadableError,
        errors,
      })
      err.message = `Function ${operationId} returned invalid data.`
      logError(err, {
        highlightedWords: [operationId]
      })
      
      
```

You probably have to import GleeError above. If I didn't commit any mistake 😅  that should give a better hint to the user on what actually went wrong.

**Review by fmvilas — CHANGES_REQUESTED** [review]

**Comment by sudoshreyansh** [comment]
@fmvilas I have made the changes.

**Review by fmvilas — APPROVED** [review]
Great stuff 👏

**Pull request merged**